### ISSUE TITLE
[Bug]: Shadows are not displayed correctly

### ISSUE THREAD

**Issue by iberezansky** [mozilla/pdf.js#18755]
### Attach (recommended) or Link to PDF file

[Heft-2-2425-2.pdf](https://github.com/user-attachments/files/17021921/Heft-2-2425-2.pdf)

### Web browser and its version

Safari 17.6

### Operating system and its version

MacOS 14.6.1

### PDF.js version

4.6.131

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

Open the attached PDF in Safari using https://mozilla.github.io/pdf.js/web/viewer.html

### What is the expected behavior?

Rendering gradient shadows

### What went wrong?

Rendering grey bars instead of shadows

### Additional context

This is a regression issue. 2.5.207 renders shadows without issues.

**Comment by Snuffleupagus** [comment]
> ### Web browser and its version
> Safari 17.6
> ### Operating system and its version
> MacOS 14.6.1

WFM, using PDF.js 4.6.161 [f68310b7b] in Firefox Nightly 132.0a1 (2024-09-16) respectively Google Chrome Beta 129.0.6668.42 running on Windows 11.

> ### Additional context
> 
> This is a regression issue. 2.5.207 renders shadows without issues.

There's been many changes and improvements in the *four years* since that version was released, and unfortunately we've seen repeatedly that Safari often have worse support (compared to other browsers) for various web-platform features.

If this is specific to Safari that unfortunately suggests a *browser-specific* bug, rather than a general PDF.js one, since otherwise you'd expect it to be consistently broken everywhere.
Please report this to the Safari developers instead, since this unfortunately doesn't look fixable here.

**Comment by iberezansky** [comment]
Reported the issue to Safari developers https://bugs.webkit.org/show_bug.cgi?id=279812

**Comment by ADTC** [comment]
@iberezansky I think this is a duplicate of #18634.

Also I think your WebKit bug is a duplicate of [Bug 278454](https://bugs.webkit.org/show_bug.cgi?id=278454) which is actually being worked on.

**Comment by ZainabAl-khabori** [comment]
@ADTC
I don't think this is completely Safari specific, though it is by far the worst.
The shadow in the PDF @iberezansky provided was on a white background, so I don't think the whole issue was in display.
Check the following PDF: [https://www.muscatgas.com/books/company-profile/MUSCAT%20GAS_PROFILE-PRINT(HI-RES).pdf](https://www.muscatgas.com/books/company-profile/MUSCAT%20GAS_PROFILE-PRINT(HI-RES).pdf)

Just as OP said, this issue wasn't present in 2.5.207. I checked the same PDF with versions >= 3 and I noticed it as far back.
